scss/_custom-styles.scss

```diff
@@ -31,319 +31,50 @@
   color: $dark-silver;
 }
 
-//
-// >> Background Color
-//
+/*
+* >> Background Color
+*/
 
 // >>> Brand
 @each $color, $value in $colors {
   .bg-#{$color} {
-    color: $white;
+    // check for color contrast
+    @include text-color-contrast($value);
     background-color: $value;
 
-    // check for color contrast
-    @if $value == $sky or $value == $oasis or $value == $cool-gray or $value == $warm-gray or $value == $silver or $value == $leaf {
-      color: $midnight;
-      p,
-      h1,
-      h2,
-      h3,
-      h4,
-      h5,
-      h6,
-      .h1,
-      .h2,
-      .h3,
-      .h4,
-      .h5,
-      .h6 {
-        color: $midnight;
-        @each $text-color, $text-value in $colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-        @each $text-color, $text-value in $theme-colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-      }
-    }
-    @else if $value == $bloom or $value == $white {
-      color: $black;
-      p,
-      h1,
-      h2,
-      h3,
-      h4,
-      h5,
-      h6,
-      .h1,
-      .h2,
-      .h3,
-      .h4,
-      .h5,
-      .h6 {
-        color: $black;
-        @each $text-color, $text-value in $colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-        @each $text-color, $text-value in $theme-colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-      }
-    }
-    @else {
-      color: $white;
-      p,
-      h1,
-      h2,
-      h3,
-      h4,
-      h5,
-      h6,
-      .h1,
-      .h2,
-      .h3,
-      .h4,
-      .h5,
-      .h6 {
-        color: $white;
-        @each $text-color, $text-value in $colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-        @each $text-color, $text-value in $theme-colors {
-          // ensure elements using color classes aren't overwritten
-          &.text-#{$text-color} {
-            color: $text-value;
-          }
-        }
-      }
-    }
   }
 }
 
 // >>> Brand Transparent
 @each $color, $value in $colors-transparent {
   .bg-#{$color} {
+    @include text-color-contrast($value);
     @include background-transparent($value);
```

Comment:
Should we remove this line for now, and include it in a separate PR, where we can spend more time figuring out which translucent colors we want?